Essentials/src/com/earth2me/essentials/MetaItemStack.java

```diff
@@ -294,6 +295,8 @@ public void addFireworkMeta(final CommandSource sender, final boolean allowShort
                     if (colorMap.containsKey(color.toUpperCase())) {
                         validFirework = true;
                         primaryColors.add(colorMap.get(color.toUpperCase()).getFireworkColor());
+                    } else if (getColorFromRgbHexString(color) != null) {
```

Comment:
Rather than calling `getColorFromRgbHexString(color)` twice in each method, I suggest having an `else` clause (rather than `else if`) in which the return value is saved into a local variable. If it is null, through the exception; otherwise, add it to `primaryColors`. Let me know if you'd like to discuss.